src/webgpu/api/operation/sampling/filter_mode.spec.ts

```diff
@@ -9,6 +9,172 @@ TODO:
+                vec2<f32>(0.0, 50.0),
+                vec2<f32>(1.0, 0.0),
+                vec2<f32>(1.0, 50.0));
+              // draw a slanted plane in a specific way
+              let matrix : mat4x4<f32> = mat4x4<f32>(
+                vec4<f32>(-1.7320507764816284, 1.8322050568049563e-16, -6.176817699518044e-17, -6.170640314703498e-17),
+                vec4<f32>(-2.1211504944260596e-16, -1.496108889579773, 0.5043753981590271, 0.5038710236549377),
+                vec4<f32>(0.0, -43.63650894165039, -43.232173919677734, -43.18894577026367),
+                vec4<f32>(0.0, 21.693578720092773, 21.789791107177734, 21.86800193786621));
+
+              var output : Outputs;
+              output.fragUV = uv[VertexIndex];
+              output.Position = matrix * vec4<f32>(position[VertexIndex], 1.0);
+              return output;
+            }
+            `,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: this.device.createShaderModule({
+          code: `
+            @group(0) @binding(0) var sampler0 : sampler;
+            @group(0) @binding(1) var texture0 : texture_2d<f32>;
+
+            @fragment fn main(
+              @builtin(position) FragCoord : vec4<f32>,
+              @location(0) fragUV: vec2<f32>)
+              -> @location(0) vec4<f32> {
+                return textureSample(texture0, sampler0, fragUV);
+            }
+            `,
+        }),
+        entryPoint: 'main',
+        targets: [{ format: 'rgba8unorm' }],
+      },
+      primitive: { topology: 'triangle-list' },
+    });
+  }
+
+  // return the render target texture object
+  drawSlantedPlane(textureView: GPUTextureView, sampler: GPUSampler): GPUTexture {
+    // make sure it's already initialized
+    assert(this.pipeline !== undefined);
+
+    const bindGroup = this.device.createBindGroup({
+      entries: [
+        { binding: 0, resource: sampler },
+        { binding: 1, resource: textureView },
+      ],
+      layout: this.pipeline.getBindGroupLayout(0),
+    });
+
+    const colorAttachment = this.device.createTexture({
+      format: kColorAttachmentFormat,
+      size: { width: kRTSize, height: kRTSize, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: colorAttachmentView,
+          storeOp: 'store',
+          clearValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+          loadOp: 'clear',
+        },
+      ],
+    });
+    pass.setPipeline(this.pipeline);
+    pass.setBindGroup(0, bindGroup);
+    pass.draw(6);
+    pass.end();
+    this.device.queue.submit([encoder.finish()]);
+
+    return colorAttachment;
+  }
+}
+
+export const g = makeTestGroup(SamplerFilterModeSlantedPlaneTest);
+
+g.test('mipmapFilter')
```

Comment:
Hmm, I couldn't find a proper way to test the filter modes yet. I aimed to write a test that filter mode affected the expected colors in drawing the slanted plane. But, the filter modes(`mipmapFilter` first for now) didn't affect the expected result. Could you please give me any idea or comment on how to test the `fooFilter` affects the expected color?

Reply:
It would be useful to visualize the colorAttachment. Unfortunately we never managed to add good facilities for this - I think I started writing some but never finished.

When running in a browser, you can hack in code to visualize it, e.g. add a canvas to the `document.body` and change drawSlantedPlane to get its attachment from getCurrentTexture instead of createTexture. Example:

https://github.com/Gyuyoung/cts/compare/filter-mode-minFilter...kainino0x:cts:example-visualization
Here you can see there is a slight difference in the result (but it's subtle):

![image](https://user-images.githubusercontent.com/606355/209736136-f499c702-4519-4bda-9d9a-e636791f6ae9.png) linear
![image](https://user-images.githubusercontent.com/606355/209736129-faca005e-bf54-42a9-8957-9ac478e55d0c.png) nearest

Reply:
Thank you for giving me more information. I don't find a way to test it yet. Let me take a look at this more.

Reply:
FYI https://github.com/gpuweb/cts/issues/2072#issuecomment-1371216729